cm-vault/vault-config/main.tf

```diff
@@ -0,0 +1,43 @@
+terraform {
+  required_providers {
+    vault = {
+      source = "hashicorp/vault"
+      version = "~> 3.5.0"
+    }
+  }
+}
+
+provider "vault" {
+  address = "http://127.0.0.1:30200"
```

Comment:
This will not work  by default. I think we will need to add something to `vault-install` module to port forward, see i.e https://github.com/hashicorp/terraform-provider-kubernetes/issues/812#issuecomment-1027525534

Reply:
> And why This will not work by default., I configure that the service is NodePort type, it can be run my test env.

I can run it on a kind cluster if I create extra port mappings https://kind.sigs.k8s.io/docs/user/configuration/#extra-port-mappings.
But I think in long term, we want to make sure that it is possible to deploy for example on GKE or some other Kubernetes.

I think maybe for this PR we can add a warning that this only works with kind and add the required kind config to get this to work:
```
apiVersion: kind.x-k8s.io/v1alpha4
kind: Cluster
nodes:
- role: control-plane
  extraPortMappings:
  - containerPort: 30200
    hostPort: 30200
```

But after this pR we will probably need to set it up in such a way that this can be used on any Kubernetes cluster and does not require custom setup

Reply:
> I can run it on a kind cluster if I create extra port mappings https://kind.sigs.k8s.io/docs/user/configuration/#extra-port-mappings.

Thanks for pointing this problem, I hadn't tested it in kind cluster. I just tested it in kind cluster environment and it also happended, I am thinking how to resolve it.

Reply:
I think for now it is ok to ask everyone to deploy the kind cluster with the config that I posted above- that should work.
We can look at improving this in a separate PR, what do you think?

Reply:
It makes sense. I will update it ASAP.

Reply:
Does it work for you (with that config)?